src/panels/lovelace/cards/hui-history-graph-card.ts

```diff
@@ -43,7 +43,11 @@ export class HuiHistoryGraphCard extends LitElement implements LovelaceCard {
   private _interval?: number;
 
   public getCardSize(): number {
-    return 4;
+    if (!this._config) {
+      return 0;
+    }
+    // +1 for the header
+    return (this._config.title ? 1 : 0) + this._config.entities.length * 2;
```

Comment:
This is not necessarily true, or even near true.

Sensor data can be displayed as graphs or as bars, and those have widely different heights and growth ratios.
Unfortunately, there's no way of knowing before it's too late.